### PR TITLE
Create a tools/release-all directory to contain release-all.sh and su…

### PR DESCRIPTION
**Pull request by ajfloeder**
…pporting README.md

### CODE REVIEW

File: tools/release-all/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+## Assumptions
+
+- `master/main` branch for each repository contains **tested** software and documenation and is ready to be released.
+- You've installed the GitHub CLI tool, `gh`.
+  - This tool requires a GH_TOKEN environment variable containing a `repo` scope classic token.
+
+## Usage
+
+```text
```
Comment (roehrich-hpe) [inline tools/release-all/README.md:20]: Don't include the -h output here. That just becomes a pain in the butt later.

File: tools/release-all/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      ./release-all.sh -P release-push -R <repo>
+
+        Else if **Create Trial Release Branch** was unable to auto merge, manually merge and push the release branch
+
+        ```bash
+        cd workingspace/repo
+        # Manually merge the changes from master/main to the release branch
+        go mod tidy
+        go mod vendor
+        git commit -s # take the default commit message, don't bother editing it.
```
Comment (roehrich-hpe) [inline tools/release-all/README.md:86]: Before the commit run `git status` to confirm that all issues have been addressed.

**Review by roehrich-hpe — APPROVED** [review]

File: tools/release-all/README.md (hunk 2 above)
Comment (roehrich-hpe) [reply tools/release-all/README.md:86]: Also, this needs a `git add ....` of affected files.

**Comment by roehrich-hpe** [comment]
Update this part of `.gitignore`:

```
# For the release-all.sh tool.
tools/workingspace/
workingspace/
```
Add `tools/release-all/workingspace`.
Leave the others in place, because I don't tend to 'cd' down into the tool directory like you all do.

File: tools/release-all/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  - This tool requires a GH_TOKEN environment variable containing a `repo` scope classic token.
+
+## Steps
+
+### Run the steps in this order
+
+> **Note:** You almost always want to use the -R option to focus the `phase` activity to a specific repo.
+
+0. **List Repos:** Get the ordered list of repo names to use with -R option in subsequent steps. This is referred to as `repo-list`
+    > **Pro tip** Keep this list in a separate window for easy viewing
```
Comment (bdevcich) [inline tools/release-all/README.md:24]:
```suggestion
    > **Pro tip:** Keep this list in a separate window for easy viewing
```

File: tools/release-all/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        go mod vendor
+        git status # confirm all issues have been address
+        git add <all affected files>
+        git commit -s # take the default commit message, don't bother editing it.
+        git push --set-upstream origin <release branch name>
+        ```
+
+    2. Create PR for the pushed release branch:
+      ./release-all.sh -P create-pr -R <repo>
+    3. Merge PR for the pushed release branch: **Note: Do NOT manually merge the PR, let `release-all.sh` do it**
```
Comment (bdevcich) [inline tools/release-all/README.md:63]: Same here:

```suggestion
    2. Create PR for the pushed release branch:

        ```bash
        ./release-all.sh -P create-pr -R <repo>
        ```

```

File: tools/release-all/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      ./release-all.sh -P merge-pr -R <repo>
+    4. Tag the release:
```
Comment (bdevcich) [inline tools/release-all/README.md:65]:
```suggestion
    3. Merge PR for the pushed release branch: **Note: Do NOT manually merge the PR, let `release-all.sh` do it:**

        ```bash
        ./release-all.sh -P merge-pr -R <repo>
        ```
```

**Review by bdevcich — APPROVED** [review]

**Pull request merged**